### PR TITLE
integration otel to http clients

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: pkg/security/jwt/validator_test.go [hunk 1]
```diff
@@ -102,7 +118,11 @@ func TestInvalidToken(t *testing.T) {
 	server := newTestJwks()
 	defer server.Close()
 
-	v := jwt.NewValidator(server.URL+uri, &noTLS)
+	transport := http.DefaultTransport.(*http.Transport).Clone()
```
Comment (Danielius1922) [inline pkg/security/jwt/validator_test.go]: Could turn these 5 lines into a function instead of C&P it everywhere

Comment (jkralik) [reply pkg/security/jwt/validator_test.go]: Fixed

File: test/oauth-server/test/test.go [hunk 2]
```diff
@@ -223,6 +224,22 @@ func GetDefaultAccessToken(t *testing.T) string {
 	return GetAccessToken(t, config.OAUTH_SERVER_HOST, ClientTest)
 }
 
+func GetJWTValidator(jwkURL string) *jwt.Validator {
+	t := http.DefaultTransport.(*http.Transport).Clone()
+	//t.MaxIdleConns = 100
```
Comment (Danielius1922) [inline test/oauth-server/test/test.go]: Remove or uncomment?

Comment (jkralik) [reply test/oauth-server/test/test.go]: Fixed

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**